From acc47b01f573b1cfa94a4765112fbd04d85871e0 Mon Sep 17 00:00:00 2001
From: Markos Chandras <markos.chandras@imgtec.com>
Date: Wed, 24 Jul 2013 11:33:05 +0000
Subject: [PATCH 101/105] MIPS: EVA: Fix encoding for the UUSK AM bits on the SegCtl registers.

The UUSK AM encoding is 0b111 instead of 0b110. 0b110 is reserved
and should not be used. Also use consistent spaces in the proc file
and fix a checkpatch warning.

Signed-off-by: Markos Chandras <markos.chandras@imgtec.com>
---
 arch/mips/include/asm/mipsregs.h |    2 +-
 arch/mips/kernel/segment.c       |    5 +++--
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/arch/mips/include/asm/mipsregs.h b/arch/mips/include/asm/mipsregs.h
index f98431c..91e7b5b 100644
--- a/arch/mips/include/asm/mipsregs.h
+++ b/arch/mips/include/asm/mipsregs.h
@@ -685,7 +685,7 @@
 #define MIPS_SEGCFG_C_SHIFT	0
 #define MIPS_SEGCFG_C		(_ULCAST_(7) << MIPS_SEGCFG_C_SHIFT)
 
-#define MIPS_SEGCFG_UUSK	_ULCAST_(6)
+#define MIPS_SEGCFG_UUSK	_ULCAST_(7)
 #define MIPS_SEGCFG_USK		_ULCAST_(5)
 #define MIPS_SEGCFG_MUSUK	_ULCAST_(4)
 #define MIPS_SEGCFG_MUSK	_ULCAST_(3)
diff --git a/arch/mips/kernel/segment.c b/arch/mips/kernel/segment.c
index c59da1e..c3ceb77 100644
--- a/arch/mips/kernel/segment.c
+++ b/arch/mips/kernel/segment.c
@@ -16,8 +16,9 @@ static void proc_build_segment_config(char *str, unsigned int cfg)
 {
 	unsigned int am;
 	int len = 0;
-	static const char *am_str[] =
-		{"UK   ", "MK   ", "MSK  ", "MUSK ", "MUSUK", "USK  ", "UUSK "};
+	static const char * const am_str[] = {
+		"UK  ", "MK  ", "MSK  ", "MUSK  ", "MUSUK  ", "USK  ",
+		"*Reserved*  ", "UUSK  "};
 
 	/* Segment access mode. */
 	am = (cfg & MIPS_SEGCFG_AM) >> MIPS_SEGCFG_AM_SHIFT;
-- 
1.7.1

